lib/src/map/widget.dart

```diff
@@ -1,50 +1,129 @@
+import 'package:flutter_map/src/map/controller/map_controller.dart';
 import 'package:flutter_map/src/map/inherited_model.dart';
-import 'package:flutter_map/src/map/internal_controller.dart';
-import 'package:flutter_map/src/map/map_controller.dart';
-import 'package:flutter_map/src/map/map_controller_impl.dart';
 import 'package:flutter_map/src/map/options/options.dart';
 
-/// Renders an interactive geographical map as a widget
+/// An interactive geographical map
 ///
 /// See the online documentation for more information about set-up,
 /// configuration, and usage.
 @immutable
 class FlutterMap extends StatefulWidget {
-  /// Renders an interactive geographical map as a widget
+  /// Creates an interactive geographical map
   ///
-  /// See the online documentation for more information about set-up,
-  /// configuration, and usage.
+  /// See the properties and online documentation for more information about
+  /// set-up, configuration, and usage.
   const FlutterMap({
     super.key,
+    this.mapController,
     required this.options,
-    this.children = const [],
+    required this.children,
+    @Deprecated(
+      'Append all of these children to `children`. '
+      'This property has been removed to simplify the way layers are inserted '
+      'into the map, and allow for greater flexibility of layer positioning. '
+      'This property is deprecated since v6.',
+    )
     this.nonRotatedChildren = const [],
-    this.mapController,
   });
 
-  /// Layers/widgets to be painted onto the map, in a [Stack]-like fashion
+  /// Creates an interactive geographical map
+  ///
+  /// This constructor is a shortcut intended for only the most simple of maps.
+  /// It does not support customization of the underlying [TileLayer], and lacks
+  /// the ability to add feature layers (except for an attribution layer) or
+  /// attach a [MapController]. Use the standard constructor if these are
+  /// required.
+  ///
+  /// See the properties and online documentation for more information about
+  /// set-up, configuration, and usage.
+  ///
+  /// ---
+  ///
+  /// Provide a standard slippy map URL template to the [urlTemplate] argument,
+  /// with `{x}`, `{y}`, and `{z}` placeholders. Subdomain support is not
+  /// supported through this simple constructor.
+  ///
+  /// Provide the application's correct package name, such as 'com.example.app',
+  /// to the [userAgentPackageName] argument, to allow the tile server to
+  /// identify your application. For more information, see
+  /// [TileLayer.tileProvider]'s documentation.
+  ///
+  /// It is recommended to provide a [RichAttributionWidget] or
+  /// [SimpleAttributionWidget] to the [attribution] argument. For more
+  /// information, see their documentation.
+  FlutterMap.simple({
+    super.key,
+    required this.options,
+    required String urlTemplate,
+    required String userAgentPackageName,
+    AttributionWidget? attribution,
+  })  : children = [
```

Comment:
Instead of having a dedicated constructor which could be a bother for future evolution on the main `FlutterMap()` constructor, I would recommend to have a separated `SimpleFlutterMap` or `SimpleMap` widgets that would abstract the logic of `FlutterMap()`.

Reply:
I've removed the simple constructor altogether in e991671. It was an experiment, and I'm not really sure I like it that much.